test/functional/configuration/configuration.spec.js

```diff
@@ -312,6 +313,30 @@ test("shows error for name that matches key on window", async t => {
   await instances[0].nameField.expectError(t);
 });
 
+test("shows a warning when name is changed on existing configuration", async t => {
+  await extensionViewController.init(
+    t,
+    Object.assign({}, defaultInitInfo, {
+      settings: {
+        instances: [
+          {
+            name: "alloy",
+            propertyId: "PR123"
+          }
+        ]
+      }
+    })
+  );
+  await instances[0].nameField.typeText(t, "123");
+  await instances[0].nameChangeAlert.expectExists(t);
+});
+
+test("does not show a warning when name is changed on new configuration", async t => {
+  await extensionViewController.init(t, defaultInitInfo);
+  await instances[0].nameField.typeText(t, "123");
+  await instances[0].nameChangeAlert.expectNotExists(t);
```

Comment:
How does it know how long it has to wait before deciding that it doesn't exist?

Reply:
TestCafe has a built-in polling mechanism with a timeout: https://devexpress.github.io/testcafe/documentation/test-api/built-in-waiting-mechanisms.html

Reply:
In other words, if the element exists, it keeps polling until the element either doesn't exist (therefore the assertion passes) or the timeout is reached (therefore the assertion fails).

Reply:
So if the element doesn't exist from the get go, it automatically passes? I guess there's no way to know if it appears later on.

Reply:
Yeah, that's right.